tools/targets.py

```diff
@@ -69,6 +69,8 @@ def generate(env):
 
         if env["optimize"] == "debug" or env["optimize"] == "none":
```

Comment:
Normal debug build probably should use MT / MD based on static flag. The engine has a separate `debug_crt` flag for the MDd which is not used by default (not sure if there's any use for it besides debugging CRT related issues).

Reply:
Yeah we shouldn't use `/MDd` unless there's a strong reason to.

The flags should be kept in sync with upstream Godot ideally:

https://github.com/godotengine/godot/blob/master/SConstruct#L574-L607

https://github.com/godotengine/godot/blob/master/platform/windows/detect.py#L349-L356

Reply:
Ah, ok, thanks! The scope of this PR may be wandering just the tiniest bit :-) but I've gone ahead and synchronized the optimization flags along with the "use_static_cpp" bits from upstream Godot